tracer/src/Datadog.Trace/FeatureFlags/FeatureFlagsEvaluator.cs

```diff
@@ -0,0 +1,704 @@
+// <copyright file="FeatureFlagsEvaluator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+using Datadog.Trace.FeatureFlags.Rcm.Model;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.FeatureFlags
+{
+    internal class FeatureFlagsEvaluator
+    {
+        internal static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(FeatureFlagsEvaluator));
+
+        private static readonly string[] DateFormats =
+        {
+            "MM-dd-yyyy HH:mm:ss",
+            "MM/dd/yyyy HH:mm:ss",
+            "yyyy-MM-dd HH:mm:ss",
+            "yyyy/MM/dd HH:mm:ss",
+            "yyyy-MM-dd'T'HH:mm:ss.fff'Z'",
+            "yyyy-MM-dd'T'HH:mm:ss'Z'",
+            "MM-dd-yyyy'T'HH:mm:ss.fff'Z'",
+            "MM-dd-yyyy'T'HH:mm:ss'Z'",
```

Comment:
minor: we only ever send RFC 3339 timestamps:
```suggestion
            "yyyy-MM-dd'T'HH:mm:ss.fffK",
```